File: mobile/src/wirelessGecko/java/com/siliconlabs/bledemo/rangetest/RangeTestMode.java
package com.siliconlabs.bledemo.rangetest;

/**
 * @author Comarch S.A.
 */

public enum RangeTestMode {
    Rx(1), Tx(2);

    private final int code;

    RangeTestMode(int code) {
        this.code = code;
    }

    public int getCode() {
        return code;
    }

    public static RangeTestMode fromCode(int code) {
        switch (code) {
            case 1:
                return Rx;
            case 2:
                return Tx;
            default:
                throw new IllegalArgumentException("No mode for code: " + code);
        }
    }
}

File: mobile/src/main/java/com/siliconlabs/bledemo/RangeTest/Models/RangeTestMode.kt
package com.siliconlabs.bledemo.RangeTest.Models

/**
 * @author Comarch S.A.
 */
enum class RangeTestMode(val code: Int) {
    Rx(1),
    Tx(2);

    companion object {
        fun fromCode(code: Int): RangeTestMode {
            return when (code) {
                1 -> Rx
                2 -> Tx
                else -> throw IllegalArgumentException("No mode for code: $code")
            }
        }
    }

}